sparse/compressed/indexing.py

```diff
@@ -0,0 +1,149 @@
+import numpy as np
+import numba
+import copy
+from numbers import Integral
+from collections.abc import Iterable 
+from ..slicing import normalize_index
+from ..coo.common import linear_loc
+from ..coo.core import COO
+from functools import reduce
+from operator import mul
+from .convert import convert_to_flat,uncompress_dimension
+
+
+def getitem(x,key):

```

Comment:
You might need to refactor this function, put the critical loopy-bits into Numba as well.

Reply:
Yeah, this function is a bit messy right now. These loops just iterate over the dimensions so in an extreme case case there might be 7 iterations. This is to say that I don't think account for much of the run time. I can jit them though if you'd like.